docs/overview/whatisvpp/softwarearchitecture.rst

```diff
@@ -0,0 +1,707 @@
+.. _softwarearchitecture:
+
+.. toctree::
+
+Software Architecture
+=====================
+
+The fd.io vpp implementation is a third-generation vector packet
+processing implementation specifically related to US Patent 7,961,636,
+as well as earlier work. Note that the Apache-2 license specifically
+grants non-exclusive patent licenses; we mention this patent as a
+point of historical interest.
+
+For performance, the vpp dataplane consists of a directed graph of
```

Comment:
I would insert an Overview section header here, see my comment above the "Vppinfra" section below.